test/unittests/evm_test.cpp

```diff
@@ -630,6 +630,35 @@ TEST_P(evm, invalid)
     EXPECT_EQ(result.gas_left, 0);
 }
 
+TEST_P(evm, inner_stop)
+{
+    const auto code = push(0) + OP_STOP + OP_POP;
+    execute(3, code);
+    EXPECT_GAS_USED(EVMC_SUCCESS, 3);
+}
+
+TEST_P(evm, inner_return)
+{
+    const auto code = ret(0, 0) + push(0);
+    execute(6, code);
```

Comment:
Random question, why don't we make this also to be in the form of:
```
const auto code = ret(0, 0) + push(0;
assert(code == "60006000fd6000");
execute(code;
```

And have some script populate the assertion haha?

Reply:
In C++20 there is `constexpr std::string` so you can generate the code at compile time... Not sure how to easily script it inject hex strings to the source files.